drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodListDialogFragment.kt

```diff
@@ -199,4 +202,48 @@ class PaymentMethodListDialogFragment :
             }
         }
     }
+
+    private fun showStoredConfirmation(storedPaymentMethod: StoredPaymentMethod) {
+        AlertDialog.Builder(requireContext())
+            .setTitle(
+                String.format(
+                    resources.getString(R.string.checkout_stored_payment_confirmation_message),
+                    storedPaymentMethod.name ?: ""
+                )
+            )
+            .setNegativeButton(R.string.checkout_stored_payment_confirmation_cancel_button) { dialog, _ ->
+                dialog.dismiss()
+            }
+            .setPositiveButton(
+                PayButtonFormatter.getPayButtonText(
+                    amount = dropInViewModel.amount,
+                    locale = dropInViewModel.dropInConfiguration.shopperLocale,
+                    localizedContext = requireContext(),
+                )
+            ) { dialog, _ ->
+                dialog.dismiss()
+                onConfirmationButtonClicked(storedPaymentMethod)
+            }
+            .show()
+    }
+
+    private fun onConfirmationButtonClicked(storedPaymentMethod: StoredPaymentMethod) {
+        val component = getComponentFor(
```

Comment:
We need to pass a key here to create a new component per stored payment methods (you can use `storedPaymentMethod.id`).
Otherwise suppose we have 2 stored blik, if we click on the first one then cancel then click on the 2nd one we will still get the same component as the 1st.

Reply:
Good point 👍